### PR TITLE
Added my (open access) book

### PR DESCRIPTION
**Pull request by rougier**
Added entry to my (open access) book.
I did not test build (because it's quite simple addition).

### CODE REVIEW

File: doc/users/resources/index.rst [hunk 1]
```diff
@@ -11,6 +11,10 @@ External resources
 Books, chapters and articles
 ============================
 
+* `Scientific Visualization: Python + Matplotlib (2021)
+   <https://hal.inria.fr/hal-03427242/>`
```
Comment (story645) [inline doc/users/resources/index.rst:15]: Could the missing underscore at the end be why the docs aren't building?

**Review by story645 — CHANGES_REQUESTED** [review]

File: doc/users/resources/index.rst (hunk 1 above)
Comment (rougier) [reply doc/users/resources/index.rst:15]: Oh s**t, sorry for that. I'll correct it.

Comment (rougier) [reply doc/users/resources/index.rst]: Lesson learned. Even a simple PR needs to be tested.

Comment (story645) [reply doc/users/resources/index.rst]: there's a white space failure 
image
and it's still not building correctly :/

**Comment by jklymak** [comment]
@rougier unfortunately, its still not passing CI...

File: doc/users/resources/index.rst [hunk 2]
```diff
@@ -11,6 +11,10 @@ External resources
 Books, chapters and articles
 ============================
 
+* `Scientific Visualization: Python + Matplotlib (2021)
+   <https://hal.inria.fr/hal-03427242/>`_
+   by Nicolas P. Rougier
```
Comment (timhoffm) [inline doc/users/resources/index.rst:16]:
```suggestion
* `Scientific Visualization: Python + Matplotlib (2021)
  <https://hal.inria.fr/hal-03427242/>`_
  by Nicolas P. Rougier
```

**Comment by rougier** [comment]
Ah damn, I pressed `commit changes` thinking it would add @timhoffm proposed fix to my repor but I think I've committed my changes to masted instead. And the test is still failing. Sorry for all the trouble.

**Comment by jklymak** [comment]
Huh, now the circle CI didn't run!  I'll close and reopen...

**Review by story645 — APPROVED** [review]

**Comment by story645** [comment]
🥳 it builds, but I'm not sure if you're supposed to rebase this into one commit before this gets merged

**Pull request merged**